roles/docker-garbage-collection/templates/garbage-collect-docker.sh.j2

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# garbage-collect-docker.sh
+# Backs up and removes dangling docker images. Removes stopped containers
+#   preventing any image removal
+
+stopped_container_sed_pattern="{{stopped_container_sed_pattern}}"
+backup_images="{{backup_images}}"
+
+{% raw %}
+
+###
+# remove_image()
+# Description: Attempts to remove an image. If a stopped container prevents
+#   this, it removes the container and tries again.
+# Parameters: $1 is image ID
+###
+remove_image()
+{
+    local id
+    local removal_output
+    local removal_exit
+    id=$1
+    removal_output=$(docker rmi $id 2>&1)
+    removal_exit=$(echo $?)
+
+    # check for stopped container preventing removal
+    container_id=$(echo $removal_output | sed -n -e "$stopped_container_sed_pattern")
+    if [ $removal_exit -eq 1 ] && [ ${#container_id} -gt 1 ]
```

Comment:
I think I didn't really understand this criterion:

```shell
[ ${#container_id} -gt 1 ]
```

what does `1` stands for here?

Reply:
This condition tested the length of the container_id variable, to see if the variable has been populated. I will use -n to test if it is not empty instead.